Code/Framework/AzCore/AzCore/Serialization/Json/MapSerializer.cpp

```diff
@@ -322,7 +322,7 @@ namespace AZ
         {
             if (context.ShouldKeepDefaults() && keyValues.Empty())
             {
-                outputValue.SetArray();
+                outputValue.SetObject();
```

Comment:
This can only work for a map of unique keys(map, unordered_map)
This doesn't work for multimap and unordered_multimap

Reply:
Actually, it looks like not all multimaps would always result in an array : https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/Serialization/Json/MapSerializerTests.cpp#L852-L854

So I guess this boils down to what we want our default empty map to be. Since maps with string keys are more common for storing json data, I think setting it to {} would be better than []

Reply:
Well it was mentioned that default map is used for merging.
Merging an object over an array should replace the array contents, so having an empty json array shouldn't affect how patching is done.

Reply:
I don't think applying a patch targeting an empty array is working right now, which is why I had to introduce this change. I get the below error, which makes sense because if you have an array, you'd expect the patch to be generating targeting a path like '.../someArray/0' instead of '.../someObject'
`Reason: The operation has halted during a merge operation for '/0' because a field or element has an invalid value.`

Reply:
Discussed with @srikappa-amzn offline, but it should fine to default to serializing a map to an empty JSON object.
It would be up to the users of the rapidjson data to take care of any edge cases related to attempting to create a JSON patch from an empty C++ map<int, string> to a populated one, as well as any edges related to creating a patch from an empty C++ multimap<string, string> to a one with multiple of the same keys within it.

The multimap issue is simply that, JSON object expect keys to be unique.
This should be a problem as long a stores to a key -> array
```JSON
{
    "MultimapKeyWithMultipleElements": [ "Value1", "Value2"],
    "MultimapKeyWithOneElements": [ "Value1"]
}